kexec-tools arch purity problem

## Issue description

```
[clever@aarch64:~]$ nix-build '<nixpkgs>' -A kexectools --argstr system armv7l-linux
...
gcc -g -O2 -fno-strict-aliasing -Wall -Wstrict-prototypes -fno-zero-initialized-in-bss -mcmodel=large -fno-PIC -fno-stack-protector -fno-asynchronous-unwind-tables -Wundef -Werror-implicit-function-declaration -Wdeclaration-after-statement -Werror=implicit-int -Werror=strict-prototypes -Os -fno-builtin -ffreestanding -fno-zero-initialized-in-bss -fno-PIC -fno-PIE -fno-stack-protector -mcmodel=large -fno-PIC -fno-stack-protector -fno-asynchronous-unwind-tables -Wundef -Werror-implicit-function-declaration -Wdeclaration-after-statement -Werror=implicit-int -Werror=strict-prototypes -I./purgatory/include -I./purgatory/arch/arm64/include -I./util_lib/include -I./include -Iinclude -I/nix/store/vj20h5y4x1f7350z8rdd5a6fjzzyfs45-gcc-7.3.0/lib/gcc/armv7l-unknown-linux-gnueabihf/7.3.0/include  -c -MD -o purgatory/purgatory.o purgatory/purgatory.c
gcc: error: unrecognized command line option '-mcmodel=large'
gcc: error: unrecognized command line option '-mcmodel=large'
make: *** [Makefile:114: purgatory/purgatory.o] Error 1
builder for '/nix/store/j9zxkgp6x4c97r4j54ig53wp2ddz0mc1-kexec-tools-2.0.17.drv' failed with exit code 2
error: build of '/nix/store/j9zxkgp6x4c97r4j54ig53wp2ddz0mc1-kexec-tools-2.0.17.drv' failed
```
`kexec-tools` is auto-detecting an aarch64 host, and then passing invalid flags when doing an armv7 build

### Steps to reproduce

`nix-build '<nixpkgs>' -A kexectools --argstr system armv7l-linux` on an `aarch64-linux` box

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

All I know is (on aarch64) when I type
```
$ admin@ip-172-31-7-187:~/src/nixpkgs$ uname -a
Linux ip-172-31-7-187 4.9.0-9-arm64 #1 SMP Debian 4.9.168-1+deb9u3 (2019-06-16) aarch64 GNU/Linux
$ admin@ip-172-31-7-187:~/src/nixpkgs$ nix --version
nix (Nix) 2.2.2
$ admin@ip-172-31-7-187:~/src/nixpkgs$ git branch
  master
* release-19.03
$ nix-build '<nixpkgs>' -A kexectools --argstr system armv7l-linux --verbose
...
gcc -g -O2 -fno-strict-aliasing -Wall -Wstrict-prototypes -fno-zero-initialized-in-bss -mcmodel=large -fno-PIC -fno-stack-protector -fno-asynchronous-unwind-tables -Wundef -Werror-implicit-function-declaration -Wdeclaration-after-statement -Werror=implicit-int -Werror=strict-prototypes -Os -fno-builtin -ffreestanding -fno-zero-initialized-in-bss -fno-PIC -fno-PIE -fno-stack-protector -mcmodel=large -fno-PIC -fno-stack-protector -fno-asynchronous-unwind-tables -Wundef -Werror-implicit-function-declaration -Wdeclaration-after-statement -Werror=implicit-int -Werror=strict-prototypes -I./purgatory/include -I./purgatory/arch/arm64/include -I./util_lib/include -I./include -Iinclude -I/nix/store/a5x2k8n6yf77rwwdsncbnjfblyybkb3p-gcc-7.4.0/lib/gcc/armv7l-unknown-linux-gnueabihf/7.4.0/include  -c -MD -o purgatory/purgatory.o purgatory/purgatory.c
gcc: error: unrecognized command line option '-mcmodel=large'
gcc: error: unrecognized command line option '-mcmodel=large'
make: *** [Makefile:114: purgatory/purgatory.o] Error 1
builder for '/nix/store/d4qawsgv3sdb285s5z488ycvi3447xaa-kexec-tools-2.0.18.drv' failed with exit code 2
error: build of '/nix/store/d4qawsgv3sdb285s5z488ycvi3447xaa-kexec-tools-2.0.18.drv' failed
```

Nope that didn't fix it.

```
admin@ip-172-31-7-187:~/src/nixpkgs$ nix-build '<nixpkgs>' -A kexectools --argstr system armv7l-linux
...
gcc -g -O2 -fno-strict-aliasing -Wall -Wstrict-prototypes -fno-zero-initialized-in-bss -mcmodel=large -fno-PIC -fno-stack-protector -fno-asynchronous-unwind-tables -Wundef -Werror-implicit-function-declaration -Wdeclaration-after-statement -Werror=implicit-int -Werror=strict-prototypes -Os -fno-builtin -ffreestanding -fno-zero-initialized-in-bss -fno-PIC -fno-PIE -fno-stack-protector -mcmodel=large -fno-PIC -fno-stack-protector -fno-asynchronous-unwind-tables -Wundef -Werror-implicit-function-declaration -Wdeclaration-after-statement -Werror=implicit-int -Werror=strict-prototypes -I./purgatory/include -I./purgatory/arch/arm64/include -I./util_lib/include -I./include -Iinclude -I/nix/store/a5x2k8n6yf77rwwdsncbnjfblyybkb3p-gcc-7.4.0/lib/gcc/armv7l-unknown-linux-gnueabihf/7.4.0/include  -c -MD -o purgatory/purgatory.o purgatory/purgatory.c
gcc: error: unrecognized command line option '-mcmodel=large'
gcc: error: unrecognized command line option '-mcmodel=large'
make: *** [Makefile:114: purgatory/purgatory.o] Error 1
builder for '/nix/store/d9k652wmshshxkn75r0k6pvc6zh8gjwq-kexec-tools-2.0.18.drv' failed with exit code 2
error: build of '/nix/store/d9k652wmshshxkn75r0k6pvc6zh8gjwq-kexec-tools-2.0.18.drv' failed
```

but this works fine:

```
admin@ip-172-31-7-187:~/src/nixpkgs$ nix build -f '<nixpkgs>' pkgsCross.armv7l-hf-multiplatform.kexectools
```

Not working on master. Not working on release-18.03

```
admin@ip-172-31-7-187:~/src/nixpkgs$ git branch
* master
  release-19.03
admin@ip-172-31-7-187:~/src/nixpkgs$ git rev-parse HEAD
880bc93fc0ad44ea5b973e532c338afeb70d2a71
admin@ip-172-31-7-187:~/src/nixpkgs$ nix --version
nix (Nix) 2.2.2
admin@ip-172-31-7-187:~/src/nixpkgs$ nix-build '<nixpkgs>' -A kexectools --argstr system armv7l-linux --verbose
....
gcc -fno-zero-initialized-in-bss -mcmodel=large -fno-PIC -fno-stack-protector -fno-asynchronous-unwind-tables -Wundef -Werror-implicit-function-declaration -Wdeclaration-after-statement -Werror=implicit-int -Werror=strict-prototypes -Os -fno-builtin -ffreestanding -fno-zero-initialized-in-bss -fno-PIC -fno-PIE -fno-stack-protector -mcmodel=large -fno-PIC -fno-stack-protector -fno-asynchronous-unwind-tables -Wundef -Werror-implicit-function-declaration -Wdeclaration-after-statement -Werror=implicit-int -Werror=strict-prototypes -I./purgatory/include -I./purgatory/arch/arm64/include -I./util_lib/include -I./include -Iinclude -I/nix/store/vcbhaawz3b3nqmj1cqhvfw3g9l0py13y-gcc-7.4.0/lib/gcc/armv7l-unknown-linux-gnueabihf/7.4.0/include  -c -MD -o purgatory/purgatory.o purgatory/purgatory.c
gcc: error: unrecognized command line option '-mcmodel=large'
gcc: error: unrecognized command line option '-mcmodel=large'
make: *** [Makefile:114: purgatory/purgatory.o] Error 1
builder for '/nix/store/fh80a2byl9hvvjf3x1qvvgm9k14m6imf-kexec-tools-2.0.19.drv' failed with exit code 2
error: build of '/nix/store/fh80a2byl9hvvjf3x1qvvgm9k14m6imf-kexec-tools-2.0.19.drv' failed
```

--- a/pkgs/os-specific/linux/kexectools/default.nix
+++ b/pkgs/os-specific/linux/kexectools/default.nix
@@ -14,6 +14,9 @@ stdenv.mkDerivation rec {
 
   hardeningDisable = [ "format" "pic" "relro" "pie" ];
 
+  # Prevent kexec-tools from using uname to detect target, which is wrong in
+  # cases like compiling for aarch32 on aarch64
+  configurePlatforms = [ "build" "host" ];
   configureFlags = [ "BUILD_CC=${buildPackages.stdenv.cc.targetPrefix}cc" ];
   depsBuildBuild = [ buildPackages.stdenv.cc ];
   buildInputs = [ zlib ];

